packages/core/src/components/tree/tree.tsx

```diff
@@ -80,15 +92,16 @@ export class Tree<T = {}> extends React.Component<ITreeProps<T>, {}> {
         return this.nodeRefs[nodeId];
     }
 
-    private renderNodes(treeNodes: ITreeNode[], currentPath?: number[], className?: string): JSX.Element {
+    private renderNodes(treeNodes: Array<ITreeNode<T>>, currentPath?: number[], className?: string): JSX.Element {
```

Comment:
is this change necessary? i don't think it is as this component doesn't care about the `<T>` type: it's just for the consumer. let's try to keep this as minimal as possible.

Reply:
Since the method is called from the render method which passes in `this.props.contents`, the renderNodes function has to take in typed interfaces to render typed corresponding TypedTreeNodes. The problem that existed was that a TypedTree rendered nodes that would not be able to take in nodeData of the passed in type as expected.

Reply:
🤔 the tree nodes are rendered internally by the tree so we don't actually need the type information inside the component (it's only useful in event handlers).

Reply:
But since renderNodes is the function that passes down the onClick, onContextMenu, etc. handlers, it cannot create default TreeNodes unless the event handlers match up.